Sources/SwiftGraphQL/HTTP.swift

```diff
@@ -232,11 +232,16 @@ private func listen<Type, TypeLock>(
                 eventHandler(.failure(.network(error)))
             case let .success(message):
                 // Try to serialize the response.
-                if let data = message.data, let result = try? GraphQLResult(webSocketResponse: data, with: selection) {
-                    eventHandler(.success(result))
-                } else {
-                    eventHandler(.failure(.badpayload))
-                }
+                if let data = message.data {
+                    let jsonObject = try? JSONSerialization.jsonObject(with: data) as? [String: Any]
+                    if jsonObject?["type"] as? String == "data" {
```

Comment:
Could you paste an example return object that the socket sends? It seems like we could create a `GraphQLSubscriptionResponse` structure that contains the `type` property and decode it there.

Does that make sense to you?

Reply:
Yep we need to change this anyway because we'll start using the id field there too when we fix #33